src/app/core/store/documents/documents.effects.ts

```diff
@@ -110,12 +118,18 @@ export class DocumentsEffects {
         collectionId: action.payload.collectionId,
         data: action.payload.data
       };
-
       return this.documentService.updateDocument(documentDto).pipe(
         map(dto => DocumentConverter.fromDto(dto))
       );
     }),
-    map(document => new DocumentsAction.UpdateDataSuccess({document: document})),
+    withLatestFrom(this.store$),
+    flatMap(([document, state]) => {
+      const collection = state.collections.entities[document.collectionId];
+      const oldDocument = state.documents.entities[document.id];
```

Comment:
We should access store through selectors and not directly. Otherwise, we will need to change our code at many places when we make a small change in its structure.